src/signer.cc

```diff
@@ -34,7 +34,8 @@ std::string minio::signer::GetCanonicalRequestHash(
   //   CanonicalHeaders + '\n\n' +
   //   SignedHeaders + '\n' +
   //   HexEncode(Hash(RequestPayload))
-  std::string canonical_request = method + "\n" + uri + "\n" + query_string +
+  std::string new_uri = uri.empty()?"/":uri;
```

Comment:
You would need to fix it at https://github.com/minio/minio-cpp/blob/main/src/request.cc#L113, not here.